pkg/cmd/pulumi/doc_reader.go

```diff
@@ -229,25 +269,38 @@ func (r *markdownReader) InputHandler() func(event *tcell.EventKey, setFocus fun
 
 			switch event.Key() {
 			case tcell.KeyCtrlO:
-				link := r.focusedLink()
-				if anchor, ok := getDocumentAnchor(link); ok {
-					selection := r.view.Selection()
-					if r.view.SelectAnchor(anchor) && selection != nil {
-						r.backstack = append(r.backstack, selection)
-					}
-				} else {
-					if err := openInBrowser(link); err != nil {
-						r.showErrorDialog("opening issue", err)
-					}
-				}
+				r.OpenLink()
 			case tcell.KeyRune:
 				switch event.Rune() {
 				case '<':
 					if len(r.backstack) != 0 {
 						last := r.backstack[len(r.backstack)-1]
 						r.backstack = r.backstack[:len(r.backstack)-1]
-						r.view.SelectSpan(last, true)
+						if last.page != "" {
+							r.rootPages.SwitchToPage(last.page)
+						}
+						if last.span != nil {
+							r.view.SelectSpan(last.span, true)
+						}
+					}
+				case 'b':
```

Comment:
So this brings up a dialog with the back stack in it? Nice.

Reply:
With a little work to support widgets in dialog boxes, it would be possible to make this a menu...